tests/lib/rules/no-useless-assignment.js

```diff
@@ -0,0 +1,564 @@
+/**
+ * @fileoverview Tests for no-useless-assignment rule.
+ * @author Yosuke Ota
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/no-useless-assignment");
+const { RuleTester } = require("../../../lib/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+    parserOptions: { ecmaVersion: 2022, sourceType: "module" }
+});
+
+ruleTester.run("no-useless-assignment", rule, {
+    valid: [
+        `let foo = 'used';
+        export { foo };
+        console.log(foo);
+        foo = 'unused like but exported';`,
+
+        // Basic tests
+        `let v = 'used';
+        console.log(v);
+        v = 'used-2'
+        console.log(v);`,
```

Comment:
It also doesn't report when the value is overwritten before being read after an assignment:

```js
/* eslint no-useless-assignment: 2 */

let v;
v = 'unused'; // false negative?
v = 'used';
console.log(v);

```

Reply:
@mdjermanovic 
If you already have an idea to implement it please let me know 🙏

Reply:
> If you already have an idea to implement it please let me know

I don't have any specific idea at the moment, but I imagine it would be somewhat similar to rules that use code path analysis to check if something was executed in any/all preceding code paths, for example `no-this-before-super`.